docs/ui/ThemeSwitch.tsx

```diff
@@ -33,8 +42,43 @@ export const MarigoldThemeSwitch = ({
   initial,
   children,
 }: MarigoldThemeSwitchProps) => {
-  const [theme, setTheme] = useState(initial);
-  useEffect(() => setTheme(theme), [theme]);
+  let localTheme: string;
+
+  const searchParams = useSearchParams();
+  const themeQueryParam = searchParams?.get('theme');
+
+  const [theme, setTheme] = useState<string>(initial);
+  const router = useRouter();
+
+  if (typeof sessionStorage !== 'undefined') {
+    localTheme = sessionStorage.getItem('theme') as string;
+  }
+
+  const isInitialMount = useRef(true); // Ref to track initial mount
+
+  useEffect(() => {
+    // component did mount
+    if (isInitialMount.current) {
+      // Skip the effect on initial mount
+      isInitialMount.current = false;
+
+      if (themeQueryParam) {
+        setTheme(themeQueryParam);
+        sessionStorage.setItem('theme', themeQueryParam);
+      } else if (localTheme) {
+        setTheme(localTheme);
+        router.push(`?theme=${localTheme}`);
+      }
+
+      return;
+    }
+
+    setTheme(theme);
+    sessionStorage.setItem('theme', theme);
+    router.push(`?theme=${theme}`, {
+      scroll: false,
+    });
+  }, [theme]);
```

Comment:
there is a warning here in this file: `React Hook useEffect has missing dependencies: 'localTheme', 'router', and 'themeQueryParam'. Either include them or remove the dependency array`

Reply:
should be fixed now